### PR TITLE
Added methods getAttributeSetId getEntityTypeId

### PR DESCRIPTION
**Pull request by bordeo**
migrate methods getAttributeSetId getAttributeSet getEntityTypeId getEntityType

### CODE REVIEW

File: Setup/EavOptionSetup.php [hunk 1]
```diff
@@ -50,18 +51,26 @@ class EavOptionSetup
      */
     private $appState;
 
+    /**
+     * @var ResourceConnection
+     */
+    private $resurceConnection;
```
Comment (Vinai) [inline Setup/EavOptionSetup.php]: Typo: `$resurceConnection` is missing an `o` in `resource`.

File: Setup/EavOptionSetup.php [hunk 2]
```diff
@@ -50,18 +51,26 @@ class EavOptionSetup
...
+
     public function __construct(
         AttributeRepository $attributeRepository,
         AttributeOptionManagementService $attributeOptionManagementService,
         AttributeOptionFactory $attributeOptionFactory,
         AttributeOptionLabelFactory $attributeOptionLabelFactory,
-        AppState $appState
-    ) {
+        AppState $appState,
+        ResourceConnection $resurceConnection
```
Comment (Vinai) [inline Setup/EavOptionSetup.php]: Typo: `$resurceConnection` is missing an `o` in `resource` (same as above, just for the constructor argument variable name).

File: Setup/EavOptionSetup.php [hunk 3]
```diff
@@ -257,4 +267,85 @@ private function workaroundIssue1405()
...
+        return $setId;
+    }
+
+    /**
+     * Retrieve Attribute set data by id or name
+     *
+     * @param int|string $entityTypeId
+     * @param int|string $id
+     * @param string $field
+     * @return mixed
```
Comment (Vinai) [inline Setup/EavOptionSetup.php]: Maybe `@return int|string|null`? Then it's at least clear it does not return a compound type.

File: Setup/EavOptionSetup.php [hunk 4]
```diff
@@ -257,4 +267,85 @@ private function workaroundIssue1405()
...
+
+        return $entityTypeId;
+    }
+
+    /**
+     * Retrieve Entity Type Data
+     *
+     * @param int|string $id
+     * @param string $field
+     * @return mixed
```
Comment (Vinai) [inline Setup/EavOptionSetup.php]: Same for the return type here as above, I think `string|int|null` would be more helpful than `mixed`.

**Review by Vinai — CHANGES_REQUESTED** [review]
Thank you very much for your PR!  
Sorry I'm reviewing it so late, I had missed the email notification.
There are a few minor changes I would like to ask you to make, then I'm very happy to merge.

In addition to the small change requests, I noticed you sometimes use double and sometimes single quoted  strings, without any reason I could discern.  
It's not important, but it would satisfy my OCD if you would use single quote strings, unless they contain an embedded variable or `\n`.

Also, in the PHPDoc comments the capitalization of the comments is not quite consistent, e.g. `id` in one vs. `Id` in another. I don't mind merging with this inconsistencies, but I thought I'd mention them because I appreciate that level of feedback in code reviews for my own code. I hope you don't mind.

Thanks again for the PR!

**Comment by bordeo** [comment]
Hi Vinai
thank you for your review, I fix all kind of stuff. I hope now it's all good.

PS: I copied the methods from magento core changing very few things. PHPDoc and quotes remained the same.

Thank you

**Comment by bordeo** [comment]
Hi Vinai
there are other problems with my pull request?

**Review by Vinai — APPROVED** [review]

**Comment by Vinai** [comment]
I'm sorry, I just missed the update notification! Thanks for picking up the thread again!

**Pull request merged**